lib/sycamore/sycamore/transforms/standardizer.py

```diff
@@ -0,0 +1,162 @@
+            "NE": "Nebraska",
+            "NH": "New Hampshire",
+            "NJ": "New Jersey",
+            "NM": "New Mexico",
+            "NV": "Nevada",
+            "NY": "New York",
+            "OH": "Ohio",
+            "OK": "Oklahoma",
+            "OR": "Oregon",
+            "PA": "Pennsylvania",
+            "RI": "Rhode Island",
+            "SC": "South Carolina",
+            "SD": "South Dakota",
+            "TN": "Tennessee",
+            "TX": "Texas",
+            "UT": "Utah",
+            "VA": "Virginia",
+            "VT": "Vermont",
+            "WA": "Washington",
+            "WI": "Wisconsin",
+            "WV": "West Virginia",
+            "WY": "Wyoming",
+        }
+
+    def replace_abbreviations(self, input_string):
+        """
+        This method replaces the US State abbreviations with full names.
+        """
+        state_abbreviations = {key: value for key, value in self._state_abbreviations.items()}
+        abbreviations = {**state_abbreviations}
+
+        pattern = re.compile(r"\b(" + "|".join(re.escape(key) for key in abbreviations.keys()) + r")\b")
+
+        def replace_match(match):
+            return abbreviations[match.group(0)]
+
+        result_string = pattern.sub(replace_match, input_string)
+        return result_string
+
+    def standardize(self, doc: Document, key_path: List[str]) -> Document:
+        """
+        This Methods creates a new element standardises the location property of the Element
+        by replcaing all instance of US State abbreviation with correct name.
+        """
+        current = doc
+        for key in key_path[:-1]:
+            if key in current.keys():
+                current = current[key]
+            else:
+                raise KeyError(f"Key {key} not found in the dictionary among {current.keys()}")
+        target_key = key_path[-1]
+        if target_key in current.keys():
+            current[target_key] = self.replace_abbreviations(current[target_key])
+        else:
+            raise KeyError(f"Key {target_key} not found in the dictionary among {current.keys()}")
+        return doc
+
+
+class DateTimeStandardizer(Standardizer):
+    """
+    This Class standardizes the format of dateTime.
+    """
+
+    def fix_date(self, raw_dateTime):
+        """
+        This Method standardizes the datetime property of Elements by replcaing . with : and parsing date as Date
+        """
+
+        raw_dateTime = raw_dateTime.replace("Local", "")
+        raw_dateTime = raw_dateTime.replace(".", ":")
+        parsed_date = parser.parse(raw_dateTime)
+        extracted_date = parsed_date.date()
+        return raw_dateTime, extracted_date
+
+    def standardize(self, doc: Document, key_path) -> Document:
+        """
+        This Methods creates a new element with Date and standardises the dateTime property of the Element
+        """
+
+        current = doc
+        for key in key_path[:-1]:
+            if key in current.keys():
```

Comment:
Can optimize the number of lookups by doing `current.get(key, None)` and then checking if the value is None

Reply:
Think you might have missed this!